libcxx/include/support/atomic/atomic_msvc.h

```diff
@@ -435,4 +463,213 @@ _Type __host__ __atomic_fetch_min(_Type volatile *__ptr, _Delta __val, int __mem
     return __expected;
 }
 
-_LIBCUDACXX_END_NAMESPACE_CUDA
+template <typename _Tp, int _Sco>
```

Comment:
The code from here onwards repeats for GCC and MSVC, right? Can we move it to some sort of a "derived" header that is just included in both of these headers to avoid repeating all of it?

Maybe if we had a from-scratch backend for MSVC this wouldn't be an issue, but seems we aren't really in such a place, right?

Reply:
It would be 'easiest' to just strip the content from `atomic_gcc.h` and instead call that some kind of `cxx_atomic_base.h`

`atomic_gcc.h` need only include `cxx_atomic_base.h` since gcc/clang comes with the intrinsics.

`atomic_msvc.h` creates the back-end intrinsics and then includes `cxx_atomic_base.h`.

Thoughts on that method? I think that would accomplish reducing code duplication and wouldn't be too confusing. Though it still means everything relies on the GCC atomic symbols being defined.